### ISSUE TITLE
[tf.nn.ctc_loss] Why we need input *sequence_length*, when we already inputted *inputs*?

### ISSUE THREAD

**Issue by genausz** [tensorflow/tensorflow#23450]
In tf.nn.ctc_loss:

tf.nn.ctc_loss(
    labels,
    inputs,
    sequence_length,
    preprocess_collapse_repeated=False,
    ctc_merge_repeated=True,
    ignore_longer_outputs_than_inputs=False,
    time_major=True
)

_**inputs**_ are sequence,
_**sequence_length**_ are length of sequence.

Q1:
Why we need to setting **_sequence_length_** when sequence are given?
Q2:
Can we setting **_sequence_length_** shorter then len(inputs)?

**Comment by harshini-gadige** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged//tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.
